tinygrad/tensor.py

```diff
@@ -36,6 +36,7 @@ class Tensor:
 
   def __init__(self, data:Union[int, float, list, tuple, LazyBuffer, np.ndarray], device=Device.DEFAULT, dtype:Optional[DType]=None, requires_grad:Optional[bool]=None):
     assert dtype is None or isinstance(dtype, DType), f"invalid dtype {dtype}"
+    assert dtype is None or device not in dtype.excluded, f"dtype: {dtype} not supported on device: {device}"
```

Comment:
You want this after the device `canonicalize` and you only want to check before the colon.

"gpu:0", "GPU:0" can both be passed into device